libs/clients/judicial-system-sp/src/lib/judicialSystemSPClient.service.ts

```diff
@@ -0,0 +1,91 @@
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import type { Logger } from '@island.is/logging'
+import { Inject, Injectable } from '@nestjs/common'
+import {
+  CaseControllerGetAllCasesLocaleEnum,
+  CaseControllerGetCaseLocaleEnum,
+  CaseControllerGetSubpoenaLocaleEnum,
+  CaseControllerUpdateSubpoenaRequest,
+  CasesApi,
+  DefendersApi,
+} from '../../gen/fetch'
+import { Auth, AuthMiddleware, User } from '@island.is/auth-nest-tools'
+import { Locale } from 'locale'
+
+@Injectable()
+export class JudicialSystemSPClientService {
+  constructor(
+    @Inject(LOGGER_PROVIDER) private readonly logger: Logger,
+    private casesApi: CasesApi,
+    private defendersApi: DefendersApi,
+  ) {}
+
+  private casesApiWithAuth = (user: User) =>
+    this.casesApi.withMiddleware(new AuthMiddleware(user as Auth))
+
+  private defenderApiWithAuth = (user: User) =>
+    this.defendersApi.withMiddleware(new AuthMiddleware(user as Auth))
+
+  async getCases(user: User, locale: Locale) {
+    let cases
+    try {
+      cases = await this.casesApiWithAuth(user).caseControllerGetAllCases({
+        locale: locale as CaseControllerGetAllCasesLocaleEnum,
```

Comment:
Kinda scary to use `as`, is there a more type-safe conversion possible? Is there a chance the client's config model changes and the locale conversion breaks?